SonarQube.Common/CommandLine/ArgumentDescriptor.cs

```diff
@@ -14,7 +14,11 @@ namespace SonarQube.Common
     /// </summary>
     public class ArgumentDescriptor
     {
-        public static readonly StringComparer IdComparer = StringComparer.InvariantCultureIgnoreCase;
+        // https://msdn.microsoft.com/en-us/library/ms973919.aspx
+        // "[d]ata that is designed to be culture-agnostic and linguistically irrelevant should... 
+        //  use either StringComparison.Ordinal or StringComparison.OrdinalIgnoreCase..."
+        public static readonly StringComparer IdComparer = StringComparer.Ordinal;
```

Comment:
@duncanpMS p3: i have the feeling that there are too many `StringComparer` defined a little bit everywhere, in `Property`, `AnalysisSetting`, etc. etc. - whereas in the end we (always?) want to set it to `Ordinal`